packages/edit-site/src/components/dataviews/provider.js

```diff
@@ -0,0 +1,253 @@
+				orderby: 'date',
+				order: 'asc',
+			};
+			return {
+				dataViewRecords: getEntityRecords(
+					'postType',
+					'wp_dataviews',
+					dataViewQuery
+				),
+				dataViewIsLoading: isResolving( 'getEntityRecords', [
+					'postType',
+					'wp_dataviews',
+					dataViewQuery,
+				] ),
+				dataViewIsSaving: isSavingEntityRecord(
+					'postType',
+					'wp_dataviews'
+				),
+			};
+		},
+		[ typeTaxonomyId ]
+	);
+	const { saveEntityRecord } = useDispatch( coreStore );
+	useEffect( () => {
+		if (
+			dataViewRecords?.length === 0 &&
+			! dataViewIsLoading &&
+			! dataViewIsSaving &&
+			! isCreatingADefaultView.current
+		) {
+			isCreatingADefaultView.current = true;
+			saveEntityRecord( 'postType', 'wp_dataviews', {
+				title: 'All',
+				status: 'publish',
+				wp_dataviews_type: typeTaxonomyId,
+				content: JSON.stringify( DEFAULT_VIEWS[ type ] ),
+			} );
+		}
+	}, [
+		type,
+		dataViewIsLoading,
+		dataViewRecords,
+		dataViewIsSaving,
+		typeTaxonomyId,
+		saveEntityRecord,
+	] );
+	useEffect( () => {
+		if ( dataViewRecords?.length > 0 ) {
+			isCreatingADefaultView.current = false;
+		}
+	}, [ dataViewRecords ] );
+	useEffect( () => {
+		isCreatingADefaultView.current = false;
+	}, [ typeTaxonomyId ] );
+	if ( dataViewRecords?.length > 0 ) {
+		return dataViewRecords;
+	}
+	return null;
+}
+
+function DataviewsProviderInner( { type, children } ) {
+	const [ currentViewId, setCurrentViewId ] = useState( null );
+	const dataviewTypeTaxonomyId = useDataviewTypeTaxonomyId( type );
+	const dataviews = useDataviews( type, dataviewTypeTaxonomyId );
+	const { editEntityRecord } = useDispatch( coreStore );
+	useEffect( () => {
+		if ( ! currentViewId && dataviews?.length > 0 ) {
+			setCurrentViewId( dataviews[ 0 ].id );
+		}
+	}, [ currentViewId, dataviews ] );
+	const editedViewRecord = useSelect(
+		( select ) => {
+			if ( ! currentViewId ) {
+				return;
+			}
+			const { getEditedEntityRecord } = select( coreStore );
+			const dataviewRecord = getEditedEntityRecord(
+				'postType',
+				'wp_dataviews',
+				currentViewId
+			);
+			return dataviewRecord;
+		},
+		[ currentViewId ]
+	);
+
+	const value = useMemo( () => {
+		return {
+			taxonomyId: dataviewTypeTaxonomyId,
+			dataviews,
+			currentViewId,
+			setCurrentViewId,
+			view: editedViewRecord?.content
+				? JSON.parse( editedViewRecord?.content )
+				: DEFAULT_VIEWS[ type ],
+			setView( view ) {
+				if ( ! currentViewId ) {
+					return;
+				}
+				editEntityRecord( 'postType', 'wp_dataviews', currentViewId, {
+					content: JSON.stringify( view ),
+				} );
+			},
+		};
+	}, [
+		type,
+		dataviewTypeTaxonomyId,
+		dataviews,
+		currentViewId,
+		editedViewRecord?.content,
+		editEntityRecord,
+	] );
+
+	return (
+		<DataviewsContext.Provider value={ value }>
+			{ children }
+		</DataviewsContext.Provider>
+	);
+}
+export default function DataviewsProvider( { children } ) {
+	const {
+		params: { path },
+	} = useLocation();
```

Comment:
I don't believe we should couple data views component with the url path. The connection should be made with a param passed to it. How it would work if we have a view with more than one data view components? It's not the current use case, but the component is meant to be reused by consumers to just render data views, so they could have more than one instances in a page and also nothing to do with the url.